Pal/Console.Uefi.cs

```diff
@@ -0,0 +1,143 @@
+﻿namespace System
+{
+    public static class Console
+    {
+        static ushort s_consoleAttribute;
+        static ushort s_cursorX;
+        static ushort s_cursorY;
+        static ushort s_windowSizeX;
+        static ushort s_windowSizeY;
+
+        public static unsafe string Title
+        {
+            set
+            {
+            }
+        }
+
+        public static unsafe bool CursorVisible
+        {
+            set
+            {
+                EfiRuntimeHost.SystemTable->ConOut->EnableCursor(EfiRuntimeHost.SystemTable->ConOut, value);
+            }
+        }
+
+        public unsafe static ConsoleColor ForegroundColor
+        {
+            set
+            {
+                s_consoleAttribute = (ushort)value;
+                uint color = s_consoleAttribute;
+                EfiRuntimeHost.SystemTable->ConOut->SetAttribute(EfiRuntimeHost.SystemTable->ConOut, color);
+            }
+        }
+
+
+        static char lastKey = '\0';
+        static ushort lastScanCode;
+        public static unsafe bool KeyAvailable
+        {
+            get
+            {
+                if (lastKey == '\0')
```

Comment:
Should this be `!=`?

Reply:
UEFI is single threaded environment. Due to how game loop created, I would always ask for key available. Thats' would be expensive on VM side. This is just hack to first time read key always.